edgelet/kube-client/src/client.rs

```diff
@@ -293,14 +293,42 @@ where
         .flatten()
     }
 
+    pub fn delete_deployment(
+        &mut self,
+        namespace: &str,
+        name: &str,
```

Comment:
You may want to add `propagation_policy: Option<&'a str>` as a parameter.  You may also want to add `grace_period_seconds: Option<i64>`  if you think you want to give Agent time to do its work.

Reply:
We should definitely have more fine tuning done when we validate this E2E, compared to the docker runtime shutdown/termination behavior. Currently, there is no grace period defined for replace deployment/service account either. So, I will refrain from poking the beast that is default values. Besides, I think there is a deployment container configuration in the deployment spec itself called terminationGracePeriodSeconds. I think we should figure out the appropriate values there maybe?

Reply:
Yes - I was thinking you are almost certainly going to want the propagation policy, but the grace period may or may not be needed.  There's no requirement to do it now.

Reply:
This deserves a separate PR because we don't propagate in the replace case as well. 
We should review the [link for cascade deletion](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion) and spin off some work on this.